homeassistant/helpers/entity.py

```diff
@@ -151,7 +170,13 @@ def update_ha_state(self, force_refresh=False):
                 "No entity id specified for entity {}".format(self.name))
 
         if force_refresh:
-            self.update()
+            if hasattr(self, 'async_update'):
```

Comment:
@lwis it's a dangerous game. Too many things will assume they can just call `update` from anywhere (outside the loop) and thus it will break.

I am however considering replacing the default `update` method to see if there is an `async_update` method, and if it is, call the `async_update` method threadsafe. That way just defining `async_update` will work for the user.